dll/win32/shell32/folders/CRecycleBin.cpp

```diff
@@ -171,16 +123,94 @@ BOOL WINAPI CBSearchRecycleBin(IN PVOID Context, IN HANDLE hDeletedFile)
     return ret;
 }
 
-static PIDLRecycleStruct * _ILGetRecycleStruct(LPCITEMIDLIST pidl)
+/*
+ * Recycle Bin folder
+ */
+
+HRESULT CRecyclerExtractIcon_CreateInstance(
+    LPCITEMIDLIST pidl, REFIID riid, LPVOID * ppvOut)
 {
-    LPPIDLDATA pdata = _ILGetDataPointer(pidl);
+    PIDLRecycleStruct *pFileDetails = _ILGetRecycleStruct(pidl);
 
-    if (pdata && pdata->type == 0x00)
-        return (PIDLRecycleStruct*) & (pdata->u.crecycle);
+    // Try to obtain the file
+    SEARCH_CONTEXT Context;
+    Context.pFileDetails = pFileDetails;
+    Context.bFound = FALSE;
 
-    return NULL;
+    if (pFileDetails == NULL) goto error;
+
+    EnumerateRecycleBinW(NULL, CBSearchRecycleBin, (PVOID)&Context);
```

Comment:
Note that other usages of `EnumerateRecycleBinW` in this file specify an explicit drive root for the first parameter (even if it's partly hackish, since recycle bins can be found on more than just one drive, and the CRecycleBin virtual folder just makes the synthesis of all recycle bins currently available on the system -- This is yet another something not completely implemented in ROS).
So it does work here with NULL ?

Reply:
I think that function expects a root, and in case it's null it uses the default recycled, i can specify this, obtaining it from the file, tomorrow i will do that

Reply:
@HBelusca the function declaration says  * Remarks: 'pszRoot' can be NULL to mean 'all recycle bins'.
So its should not be a problem on using NULL as wel are trying to localize the file, you cannot have for example the file C:\example.txt on D:\RECYCLED as windows moves to the drive's recycled folder, so it should not be  a probelm

Reply:
ok it's fine then :)